olc

```diff
@@ -4,23 +4,32 @@ import sys
 from src.lexer import Lexer
 from src.parser import Parser
 from src.interpreter import Interpreter
-
+from src.abstract_syntax_tree import AbstractSyntaxTree
 
 class Orchestrator:
-    def __init__(self):
-        self.lexer = Lexer()
-        self.parser = Parser()
-        self.interpreter = Interpreter()
+    # def __init__(self):
+    #     self.lexer = Lexer()
+    #     self.parser = Parser()
+    #     self.interpreter = Interpreter()
 
     def execute(self, code):
-        self.lexer.load_source_code(code)
-        self.lexer.process_source_code()
-
-        self.parser.tokens = self.lexer.tokens
-        ast = self.parser.parse()
-
-        if ast:
-            return self.interpreter.interpret(ast.root)
+        lexer = Lexer(code)
+        lexer.process_source_code()
+        tokens = lexer.get_tokens()
+
+        parser = Parser(tokens)
+        statements = parser.parse()
+
+        # if statements:
+        interpreter = Interpreter()
+        try:
+            interpreter.interpret(statements)
```

Comment:
With this change the orchestrator is going to create a new lexer, parser, and interpreter for every call to execute. Do you mind helping me understand the purpose for this?